addon/utils/serialize-and-push.js

```diff
@@ -0,0 +1,10 @@
+import { isArray } from '@ember/array';
+
+export default function(options, response) {
+  let modelName = this.constructor.modelName || this.constructor.typeKey;
+  let modelClass = this.store.modelFor(modelName);
+  let serializer = this.store.serializerFor(modelName);
+  let normalized = isArray(response.data) ? serializer.normalizeArrayResponse(this.store, modelClass, response) :
+      serializer.normalizeSingleResponse(this.store, modelClass, response);
```

Comment:
That's a good suggestion. This changes work for my use-cases but I agree that it's JSON API specific and that there may be cases where those functions require additional inputs. I wonder if we can split this PR so that it just adds the `before` and `after` hooks, then we can iterate on the `serializeAndPush` feature in a separate PR.

Reply:
Yes, let's do two PRs. Thanks for the revisions.